### ISSUE TITLE
[Bug]:  Hello！After read the paper “puma”， i have a question, that is know how to test the communication time and size, thanks!!!!!!!!!!!!!!

### ISSUE THREAD

**Issue by zhangwaer** [secretflow/spu#735]
### Issue Type

Performance

### Modules Involved

SPU runtime

### Have you reproduced the bug with SPU HEAD?

Yes

### Have you searched existing issues?

Yes

### SPU Version

spu0.5.0

### OS Platform and Distribution

ubuntu18.04

### Python Version

3.10.13

### Compiler Version

GCC11.2.1

### Current Behavior?

Hello！After read the paper “puma”， i have a question, that is know how to test the communication time and size, thanks!!!!!!!!!!!!!!

### Standalone code to reproduce the issue

```Python
Hello！After read the paper “puma”， i have a question, that is know how to test the communication time and size, thanks!!!!!!!!!!!!!!
```

**Comment by anakinxc** [comment]
Hi @zhangwaer 

Please [enable](https://github.com/secretflow/spu/blob/87568346ff324b5923ce19405f5878fc30785075/examples/python/conf/3pc.json#L32) profiling when start nodes

**Comment by zhangwaer** [comment]
thanks for your answer！！！  how can i  test the communication time and cost of softmax. thanks!!!

**Comment by zhangwaer** [comment]
> thank you for your answer!!!. I have a question that has been bothering me for a long time. how can i test the communication time and cost of specific function such as softmax .   thanks very much!!!

**Comment by zhangwaer** [comment]
> Hi @zhangwaer
> 
> Please [enable](https://github.com/secretflow/spu/blob/87568346ff324b5923ce19405f5878fc30785075/examples/python/conf/3pc.json#L32) profiling when start nodes


thank you for your answer!!!. I have a question that how can i test the communication time and cost of specific function such as softmax . thanks very much!!!

**Comment by fionser** [comment]
Why not just write a python script contains only one softmax ?

**Comment by zhangwaer** [comment]
> Why not just write a python script contains only one softmax ?
Dear teacher, when i use the following code to test the runtime of softmax, i found the time is similar to plaintext softmax, hope to get some help, thanks !!! 
import time
import jax
import spu
import spu.spu_pb2 as spu_pb2
from jax.scipy.special import erf

from jax import random
import jax.numpy as jnp
import secretflow as sf
from flax import linen as nn
from spu.utils.simulation import Simulator, sim_jax
start_time = time.time()

def softmax(x):


     x_max = jnp.max(x, axis=-1, keepdims=True)
     exp_x = jnp.exp(x - x_max)
     softmax_x = exp_x / jnp.sum(exp_x, axis=-1, keepdims=True)
     return softmax_x
def gelu(x):
    return 0.5 * x * (1.0 + jnp.tanh(jnp.sqrt(2.0 / jnp.pi) * (x + 0.044715 * x**3)))

key = random.PRNGKey(0)
input_array = random.normal(key, (1, 100))  # 生成一个形状为 (1, 100) 的随机输入数组

sim = Simulator.simple(3, spu.ProtocolKind.ABY3, spu.FieldType.FM64)

spu_euclidean_distance_fn = sim_jax(sim, gelu)

spu_euclidean_distance_fn(input_array)
end_time = time.time()
time=end_time -start_time 
print(time)

**Comment by anakinxc** [comment]
> > Why not just write a python script contains only one softmax ?
> > Dear teacher, when i use the following code to test the runtime of softmax, i found the time is similar to plaintext softmax, hope to get some help, thanks !!!
> > import time
> > import jax
> > import spu
> > import spu.spu_pb2 as spu_pb2
> > from jax.scipy.special import erf
> 
> from jax import random import jax.numpy as jnp import secretflow as sf from flax import linen as nn from spu.utils.simulation import Simulator, sim_jax start_time = time.time()
> 
> def softmax(x):
> 
> ```
>  x_max = jnp.max(x, axis=-1, keepdims=True)
>  exp_x = jnp.exp(x - x_max)
>  softmax_x = exp_x / jnp.sum(exp_x, axis=-1, keepdims=True)
>  return softmax_x
> ```
> 
> def gelu(x): return 0.5 * x * (1.0 + jnp.tanh(jnp.sqrt(2.0 / jnp.pi) * (x + 0.044715 * x**3)))
> 
> key = random.PRNGKey(0) input_array = random.normal(key, (1, 100)) # 生成一个形状为 (1, 100) 的随机输入数组
> 
> sim = Simulator.simple(3, spu.ProtocolKind.ABY3, spu.FieldType.FM64)
> 
> spu_euclidean_distance_fn = sim_jax(sim, gelu)
> 
> spu_euclidean_distance_fn(input_array) end_time = time.time() time=end_time -start_time print(time)

Hi @zhangwaer 

What's the question here?

**Comment by zhangwaer** [comment]
> > > Why not just write a python script contains only one softmax ?
> > > Dear teacher, when i use the following code to test the runtime of softmax, i found the time is similar to plaintext softmax, hope to get some help, thanks !!!
> > > import time
> > > import jax
> > > import spu
> > > import spu.spu_pb2 as spu_pb2
> > > from jax.scipy.special import erf
> > 
> > 
> > from jax import random import jax.numpy as jnp import secretflow as sf from flax import linen as nn from spu.utils.simulation import Simulator, sim_jax start_time = time.time()
> > def softmax(x):
> > ```
> >  x_max = jnp.max(x, axis=-1, keepdims=True)
> >  exp_x = jnp.exp(x - x_max)
> >  softmax_x = exp_x / jnp.sum(exp_x, axis=-1, keepdims=True)
> >  return softmax_x
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > def gelu(x): return 0.5 * x * (1.0 + jnp.tanh(jnp.sqrt(2.0 / jnp.pi) * (x + 0.044715 * x**3)))
> > key = random.PRNGKey(0) input_array = random.normal(key, (1, 100)) # 生成一个形状为 (1, 100) 的随机输入数组
> > sim = Simulator.simple(3, spu.ProtocolKind.ABY3, spu.FieldType.FM64)
> > spu_euclidean_distance_fn = sim_jax(sim, gelu)
> > spu_euclidean_distance_fn(input_array) end_time = time.time() time=end_time -start_time print(time)
> 
> Hi @zhangwaer
> 
> What's the question here?

When i use the code to test the runtime of softmax, i found the time is similar to plaintext softmax, hope to get some help, thanks !!!
